src/UI/Art/SpriteArtist/OverlayArtist.java

```diff
@@ -0,0 +1,110 @@
+package UI.Art.SpriteArtist;
+
+import java.awt.AlphaComposite;
+import java.awt.Color;
+import java.awt.Composite;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import Engine.XYCoord;
+import Terrain.GameMap;
+import UI.GameOverlay;
+
+public class OverlayArtist
+{
+  private static final Color HIGHLIGHT_COLOR = new Color(255, 255, 255, 160); // white
+
+  public static final Color MOBILE_FIRE_EDGE = new Color(240,   0,   0, 255); // red
+  public static final Color SIEGE_FIRE_EDGE  = new Color(  0,   0, 160, 255); // dark blue
+  public static final Color FIRE_FILL        = new Color(  0,   0,   0,   0);
+
+  private static final int OVERLAY_EDGE_THICKNESS = 2;
+
+  static AlphaComposite buffComposite = null;
+  static long lastCompositeCreationTime = 0;
+
+  public static void drawHighlights(Graphics bigG,
+                                    GameMap gameMap,
+                                    Collection<GameOverlay> inputOverlays,
+                                    int drawX, int drawY,
+                                    int viewWidth, int viewHeight,
+                                    int tileSize)
+  {
+    if( viewWidth < 1 || viewHeight < 1 )
+      return;
+
+    BufferedImage overlayImage = SpriteLibrary.createTransparentSprite(viewWidth, viewHeight);
```

Comment:
Instead of regenerating this every frame, consider caching it and only regenerating when something changes.

Reply:
blehhhhhh

Reply:
More specifically, I can cache screen dimensions and whatever, but knowing how and when an overlay changes isn't necessarily trivial.
I think drawing only on the viewport rather than caching is the better optimization of valor here.

Reply:
Even if it's just on each cursor movement, that saves a lot of regen cycles.
Drawing only on the viewport? As in without double-buffering?
Also I don't think I've heard the term "crunchy" in this context before. I assume it just means "textured"?

Reply:
Right, but AFAIK there's no easy way of determining that something has happened to cause the overlays to be different from last time.
I suppose I could just check how many overlays there are, and listen for any and all GameEvents?

"Textured" texturing, yes.
It was an attempt to convey that the texturing is supposed to be more satisfying to look at than the stuff we have now.
Then again, simplicity on a passive visual is good.